### PR TITLE
[expo-notifications] Presenting notifications

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Another feature for `expo-notifications`.

# How

On top of `@sjchmiela/expo-notifications-building-notifications`:
- added _presenter_ modules, which basically just pass in an object-y request to builders

# Test Plan

Added the most adorable tests I've ever seen:

File

Screenshot_20200131-165736

### CODE REVIEW

**Review by mczernek — APPROVED** [review]

File: packages/expo-notifications/ios/EXNotifications/Presenting/EXNotificationPresentationModule.m [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                    notification:(NSDictionary *)notificationSpec
+                    resolve:(UMPromiseResolveBlock)resolve
+                    reject:(UMPromiseRejectBlock)reject)
+{
+  UNNotificationContent *content = [_notificationBuilder notificationContentFromRequest:notificationSpec];
+  UNNotificationTrigger *trigger = nil;
+  UNNotificationRequest *request = [UNNotificationRequest requestWithIdentifier:identifier content:content trigger:trigger];
+  [[UNUserNotificationCenter currentNotificationCenter] addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+    if (error) {
+      NSString *message = [NSString stringWithFormat:@"Notification could not have been presented: %@", error.description];
```
Comment (esamelson) [inline packages/expo-notifications/ios/EXNotifications/Presenting/EXNotificationPresentationModule.m]:
```suggestion
      NSString *message = [NSString stringWithFormat:@"Notification could not be presented: %@", error.description];
```

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/ExpoNotificationPresentationModule.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  public void onCreate(ModuleRegistry moduleRegistry) {
+    mNotificationBuilderFactory = moduleRegistry.getModule(NotificationBuilderFactory.class);
+  }
+
+  @ExpoMethod
+  public void presentNotificationAsync(String identifier, Map notificationSpec, Promise promise) {
+    if (mNotificationBuilderFactory == null) {
+      promise.reject("ERR_UNAVAILABLE", "Notification builder could not have been found.");
```
Comment (esamelson) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/ExpoNotificationPresentationModule.java]:
```suggestion
      promise.reject("ERR_NOTIFICATION_BUILDER_UNAVAILABLE", "NotificationBuilder class could not be found.");
```

File: packages/expo-notifications/src/presentNotificationAsync.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  ios?: IosNotificationRequest;
+  android?: AndroidNotificationRequest;
+};
+
+export default async function presentNotificationAsync({
+  identifier,
+  ...notification
+}: NotificationRequest): Promise<void> {
+  if (!NotificationPresenter.presentNotificationAsync) {
+    throw new UnavailabilityError('ExpoNotifications', 'presentNotificationAsync');
```
Comment (esamelson) [inline packages/expo-notifications/src/presentNotificationAsync.ts]:
```suggestion
    throw new UnavailabilityError('Notifications', 'presentNotificationAsync');
```

File: apps/test-suite/tests/NewNotifications.js [hunk 4]
```diff
@@ -54,5 +55,31 @@ export async function test(t) {
...
+          title: 'Sample title',
+          subtitle: 'What an event!',
+          message: 'An interesting event has just happened',
+          badge: 1,
+        });
+      });
+
+      t.it('presents a notification with attachments', async () => {
+        const fileUri = FileSystem.documentDirectory + 'expo-notifications-test-image.jpg';
+        await FileSystem.downloadAsync('http://placekitten.com/200/300', fileUri);
```
Comment (esamelson) [inline apps/test-suite/tests/NewNotifications.js]: 😍

**Review by esamelson — APPROVED** [review]
looks great! just a few tiny wording things.

**Pull request merged**